controllers/machinehealthcheck_controller_test.go

```diff
@@ -46,906 +45,847 @@ import (
-					if err != nil {
-						return nil
-					}
-					return mhc.GetLabels()
-				}).Should(HaveKeyWithValue(clusterv1.ClusterLabelName, cluster.Name))
-			})
-
-			Specify("when other labels are present", func() {
-				mhc.Spec.ClusterName = cluster.Name
-				mhc.Labels = map[string]string{
-					"extra-label": "1",
-				}
-				mhc.SetGenerateName("correct-cluster-name-label-when-other-labels-present-")
-				Expect(testEnv.Create(ctx, mhc)).To(Succeed())
-
-				Eventually(func() map[string]string {
-					err := testEnv.Get(ctx, util.ObjectKey(mhc), mhc)
-					if err != nil {
-						return nil
-					}
-					return mhc.GetLabels()
+		g.Eventually(func() []metav1.OwnerReference {
+			err := testEnv.Get(ctx, util.ObjectKey(mhc), mhc)
+			if err != nil {
+				fmt.Printf("error cannot retrieve mhc in ctx: %v", err)
+				return nil
+			}
+			return mhc.GetOwnerReferences()
+		}, timeout, time.Second).Should(And(
+			HaveLen(1),
+			ContainElement(ownerReferenceForCluster(ctx, g, cluster)),
+		))
+	})
 
-				}).Should(And(
-					HaveKeyWithValue(clusterv1.ClusterLabelName, cluster.Name),
-					HaveKeyWithValue("extra-label", "1"),
-					HaveLen(2),
-				))
-			})
-		})
+	t.Run("it should ensure an owner reference is present when modifying existing ones", func(t *testing.T) {
+		g := NewWithT(t)
+		ctx := context.TODO()
+		cluster := createNamespaceAndCluster(g)
+
+		mhc := newMachineHealthCheck(cluster.Namespace, cluster.Name)
+		mhc.OwnerReferences = []metav1.OwnerReference{
+			{Kind: "Foo", APIVersion: "foo.bar.baz/v1", Name: "Bar", UID: "12345"},
+		}
+
+		g.Expect(testEnv.Create(ctx, mhc)).To(Succeed())
+		defer func(do ...runtime.Object) {
+			g.Expect(testEnv.Cleanup(ctx, do...)).To(Succeed())
+		}(cluster, mhc)
+
+		g.Eventually(func() []metav1.OwnerReference {
+			err := testEnv.Get(ctx, util.ObjectKey(mhc), mhc)
+			if err != nil {
+				return nil
+			}
+			return mhc.GetOwnerReferences()
+		}, timeout, time.Second).Should(And(
+			ContainElements(
+				metav1.OwnerReference{Kind: "Foo", APIVersion: "foo.bar.baz/v1", Name: "Bar", UID: "12345"},
+				ownerReferenceForCluster(ctx, g, cluster)),
+			HaveLen(2),
+		))
+	})
 
-		Context("it should ensure an owner reference is present", func() {
-			Specify("when no existing ones exist", func() {
-				mhc.Spec.ClusterName = cluster.Name
-				mhc.OwnerReferences = nil
-				mhc.SetGenerateName("owner-reference-when-no-existing-ones-exist-")
-				Expect(testEnv.Create(ctx, mhc)).To(Succeed())
-				Eventually(func() []metav1.OwnerReference {
-					err := testEnv.Get(ctx, util.ObjectKey(mhc), mhc)
-					if err != nil {
-						fmt.Fprintf(GinkgoWriter, "error cannot retrieve mhc in ctx: %s\n", err)
-						return nil
-					}
-					return mhc.GetOwnerReferences()
-				}, timeout, time.Second).Should(And(
-					ContainElement(ownerReferenceForCluster(ctx, cluster)),
-					HaveLen(1),
-				))
-			})
+	t.Run("it marks unhealthy machines for remediation when all Machines are healthy", func(t *testing.T) {
```

Comment:
Maybe rename this "it doesn't mark anything unhealthy when all Machines are healthy"?